src/schema/v2/admin/featureFlags.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  GraphQLBoolean,
+  GraphQLEnumType,
+  GraphQLFieldConfig,
+  GraphQLID,
+  GraphQLInt,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+} from "graphql"
+import { merge, sortBy } from "lodash"
+import { ResolverContext } from "types/graphql"
+import { date } from "../fields/date"
+import { base64 } from "lib/base64"
+
+export const FeatureFlagType = new GraphQLObjectType<any, ResolverContext>({
+  name: "FeatureFlag",
+  fields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+      resolve: (data) => {
+        return base64(data.name)
+      },
+    },
+    description: {
+      type: GraphQLString,
+    },
+    environments: {
+      type: new GraphQLList(
+        new GraphQLObjectType({
+          name: "FeatureFlagEnvironments",
+          fields: {
+            name: {
+              type: GraphQLString,
+            },
+            displayName: {
+              type: GraphQLString,
+            },
+            enabled: {
+              type: GraphQLBoolean,
+            },
+          },
+        })
+      ),
+    },
+    type: {
+      type: GraphQLString,
+    },
+    variants: {
+      type: new GraphQLList(
```

Comment:
`new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(WhateverType)))`

Reply:
For real 😅 ?

Reply:
Yeah: if it's an empty array; that's still technically a non-null array of non-null members. It should never be `null` or `[null]`.